File: rules/ignexcept.js
const fs = require('fs');
const vscode = require('vscode');

var smell = {

    detect : (fileName, token) => {
        try{
            if(token.hasOwnProperty("line")) var lineno = token.line;
            if(token.hasOwnProperty("type")) var tokenType = token.type;
            if(token.hasOwnProperty("exceptionHandler")) var handler = token.exceptionHandler;
            
            const MSG = 'exception might have been suppressed'
            const WARNING_MSG = MSG+' at line '+ lineno;

            const unwantedHandlers = ['continue','pass'];
            
            if(tokenType == "exception_handle" &&  unwantedHandlers.includes(handler)) smell.triggerAlarm (fileName, MSG, lineno, WARNING_MSG);
        
        } catch(error){
            console.log(error);
        }
    },
    
    triggerAlarm: (fileName, MSG, lineno, WARNING_MSG) => {
        let backslashSplittedFilePathLength = fileName.split("/").length
        let filenameFromPath = fileName.split("/")[backslashSplittedFilePathLength - 1]
        
        vscode.window.showWarningMessage(MSG +" : "+ filenameFromPath+":"+lineno);
        console.log( "\u001b[1;31m"+"warning: "+MSG +"  location:"+ fileName+":"+lineno);
        
        fs.appendFileSync('smell-spotter/warning-logs/project_warnings.csv', fileName+" ,"+WARNING_MSG+"\n");
    }
}

module.exports = smell;

File: py-scripts/filters/ignexcept.py
class IgnoreException:
    '''This is the class for detecting ignoring exception blocks in code'''

    def __init__(self):
        self.unwanted_handlers = ['continue','pass']
        self.warning_message = 'ignoring except block'


    def detect_smell(self, token, src_file):
        try:
            if token.__contains__("line"): lineno = token["line"]
            if token.__contains__("type"): token_type = token["type"] 
            if token.__contains__("exceptionHandler"): exception_handler = token["exceptionHandler"]
            
            if token_type == "exception_handle" and exception_handler in self.unwanted_handlers: 
                self.trigger_alarm(src_file, lineno)
        
        except Exception as error: 
            print("ignore except error:"+str(error))
    

    def trigger_alarm(self, src_file, lineno):
        print(src_file +":"+ str(lineno)+" ,"+self.warning_message)
    